File: src/wifi_detector_service.rb
require 'ruboto'

java_import 'android.util.Log'

class WifiDetectorService
  def onStartCommand(intent, flags, startId)
    Log.i('WifiDetector', 'Service command started')
    @receiver = Java::no.jujutsu.android.oppmote.WifiReceiver.new
    filter = Java::android.content.IntentFilter.new(Java::android.net.wifi.WifiManager::NETWORK_STATE_CHANGED_ACTION)
    registerReceiver(@receiver, filter)
    Java::android.app.Service::START_STICKY
  end

  def onDestroy
    Log.i('WifiDetector', 'Service destroyed')
    unregisterReceiver(@receiver)
    @receiver = nil
  end
end

File: src/no/jujutsu/android/oppmote/WifiDetectorService.java
package no.jujutsu.android.oppmote;

public class WifiDetectorService extends org.ruboto.RubotoService {
	public void onCreate() {
	    System.out.println("WifiDetectorService.onCreate()");
		setScriptName("wifi_detector_service.rb");
		super.onCreate();
	}

}
